default/python/AI/ProductionAI.py

```diff
@@ -1812,3 +1724,204 @@ def get_number_of_existing_outpost_and_colony_ships() -> int:
     num_colony_fleets = len(FleetUtilsAI.get_empire_fleet_ids_by_role(MissionType.COLONISATION))
     num_outpost_fleets = len(FleetUtilsAI.get_empire_fleet_ids_by_role(MissionType.OUTPOST))
     return num_outpost_fleets + num_colony_fleets
+
+
+def _try_enqueue(
+    building_type: BuildingType, candidates: Iterable[PlanetId], *, at_front: bool = False, ignore_dislike: bool = False
+) -> float:
+    """
+    Enqueue building at one of the planets in candidates.
+    If at_front, building is added at the front of the queue.
+    Returns PP per turn spent on the new building or 0.0 if nothing was enqueued.
+    """
+    universe = fo.getUniverse()
+    empire = fo.getEmpire()
+    opinion = building_type.get_opinions()
+    locations = []
+    preferred_locations = []
+    for pid in candidates:
+        planet = universe.getPlanet(pid)
+        if not planet:
+            error(f"Got pid {pid} in candidate, which does not seem to be a planetID")
+            continue
+        if not ignore_dislike and pid in opinion.dislikes:
+            continue
+        if not building_type.can_be_produced(pid) or not building_type.can_be_enqueued(pid):
+            continue
+        if pid in opinion.likes:
+            preferred_locations.append((planet.currentMeterValue(fo.meterType.maxTroops), pid))
```

Comment:
Let's extract `planet.currentMeterValue(fo.meterType.maxTroops)` to a function `_location_value(planet)`